### PR TITLE
[tap dance] restore each_tap functionality

### PR DESCRIPTION
**Pull request by pvinis**
i will soon push my own keymap where i am using that.

### CODE REVIEW

File: quantum/process_keycode/process_tap_dance.h [hunk 1]
```diff
@@ -39,15 +39,12 @@ typedef struct
...
 #define ACTION_TAP_DANCE_DOUBLE(kc1, kc2) { \
     .fn = { NULL, qk_tap_dance_pair_finished, qk_tap_dance_pair_reset }, \
-    .user_data = (void *)&((qk_tap_dance_pair_t) { kc1, kc2 })  \
+    .user_data = (void *)&((qk_tap_dance_pair_t) { kc1, kc2 }),  \
   }
 
-#define ACTION_TAP_DANCE_FN(user_fn) {  \
-    .fn = { NULL, user_fn, NULL } \
-  }
-
```
Comment (algernon) [inline quantum/process_keycode/process_tap_dance.h]: Please don't remove `ACTION_TAP_DANCE_FN`, it is a useful macro to simplify cases that only need an `on_finished` callback (I'm using it in my keymap, and while using `ACTION_TAP_DANCE_FN_ADVANCED` would be similarly easy, there is no harm in having a simpler version too.)

File: readme.md [hunk 2]
```diff
@@ -525,8 +525,7 @@ First, you will need `TAP_DANCE_ENABLE=yes` in your `Makefile`, because the feat
 This array specifies what actions shall be taken when a tap-dance key is in action. Currently, there are three possible options:
 
 * `ACTION_TAP_DANCE_DOUBLE(kc1, kc2)`: Sends the `kc1` keycode when tapped once, `kc2` otherwise. When the key is held, the appropriate keycode is registered: `kc1` when pressed and held, `kc2` when tapped once, then pressed and held.
-* `ACTION_TAP_DANCE_FN(fn)`: Calls the specified function - defined in the user keymap - with the final tap count of the tap dance action.
-* `ACTION_TAP_DANCE_FN_ADVANCED(on_each_tap_fn, on_dance_finished_fn, on_reset_fn)`: Calls the first specified function - defined in the user keymap - on every tap, the second function on when the dance action finishes (like the previous option), and the last function when the tap dance action resets.
+* `ACTION_TAP_DANCE_FN_ADVANCED(on_each_tap_fn, on_dance_finished_fn, on_dance_reset_fn)`: Calls the first specified function - defined in the user keymap - on every tap, the second function on when the dance action finishes (like the previous option), and the last function when the tap dance action resets.
```
Comment (algernon) [inline readme.md]: Like in the header, if the macro is not removed, the documentation should not be removed, either.

File: readme.md [hunk 3]
```diff
@@ -646,7 +645,7 @@ void dance_flsh_reset(qk_tap_dance_state_t *state, void *user_data) {
 qk_tap_dance_action_t tap_dance_actions[] = {
   [CT_SE]  = ACTION_TAP_DANCE_DOUBLE (KC_SPC, KC_ENT)
  ,[CT_CLN] = ACTION_TAP_DANCE_FN_ADVANCED (NULL, dance_cln_finished, dance_cln_reset)
- ,[CT_EGG] = ACTION_TAP_DANCE_FN (dance_egg)
+ ,[CT_EGG] = ACTION_TAP_DANCE_FN_ADVANCED (NULL, dance_egg, NULL)
```
Comment (algernon) [inline readme.md]: See above.

**Comment by pvinis** [comment]
alright. fixed.

**Comment by algernon** [comment]
Looks perfect now, thanks!

**Pull request merged**